File: backend/app/models/favorite.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime, String
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.models.base import Base

class Favorite(Base):
    """消息收藏模型"""
    __tablename__ = "favorites"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    message_id = Column(Integer, ForeignKey("messages.id"), nullable=False, index=True)
    note = Column(String(255), nullable=True)  # 收藏备注
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # 关系
    user = relationship("User", back_populates="favorites")
    message = relationship("Message")

File: flutter_user/lib/models/favorite/favorite.dart
import '../message/message_response.dart';

class Favorite {
  final int id;
  final int messageId;
  final int userId;
  final String? note;
  final DateTime createdAt;
  final MessageResponse? message;

  Favorite({
    required this.id,
    required this.messageId,
    required this.userId,
    this.note,
    required this.createdAt,
    this.message,
  });

  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      id: json['id'],
      messageId: json['message_id'],
      userId: json['user_id'],
      note: json['note'],
      createdAt: DateTime.parse(json['created_at']),
      message: json['message'] != null
          ? MessageResponse.fromJson(json['message'])
          : null,
    );
  }
}
